From 1ed3623876172789ce9889a9fbaf021a501e714f Mon Sep 17 00:00:00 2001
From: George McCollister <george.mccollister@gmail.com>
Date: Thu, 12 Apr 2012 12:07:08 -0500
Subject: [PATCH] reserve 512KB of free disk space

Don't install a package if it will result in less than 512KB of free
disk space.

Signed-off-by: George McCollister <george.mccollister@gmail.com>
---
 libopkg/opkg_install.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libopkg/opkg_install.c b/libopkg/opkg_install.c
index 719518f..d38d6ff 100644
--- a/libopkg/opkg_install.c
+++ b/libopkg/opkg_install.c
@@ -460,7 +460,7 @@ verify_pkg_installable(pkg_t *pkg, pkg_t *old_pkg)
 
 	kbs_available = get_available_kbytes(root_dir) + unheld_kbs;
 
-	pkg_size_kbs = (pkg->installed_size + 1023)/1024;
+	pkg_size_kbs = ((pkg->installed_size + 1023)/1024) + 512;
 
 	if (pkg_size_kbs >= kbs_available) {
 		opkg_msg(ERROR, "Only have %ldkb available on filesystem %s, "
-- 
1.7.8.4

